nanoFramework.System.Net/Sockets/Socket.cs

```diff
@@ -1042,12 +1090,18 @@ public void GetSocketOption(SocketOptionLevel optionLevel, SocketOptionName opti
         /// </remarks>
         public bool Poll(int microSeconds, SelectMode mode)
         {
-            if (m_Handle == -1)
```

Comment:
Same here. See the official documentation [here](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socket.poll?view=netframework-4.8).

Reply:
SocketException now being thrown